Sources/TuistCore/MetadataProviders/CodesignController.swift

```diff
@@ -0,0 +1,49 @@
+import FileSystem
+import Path
+import Command
+
+public final class CodesignController {
+    private let fileSystem: FileSysteming
+    private let commandRunner: CommandRunning
+
+    public init(
+        fileSystem: FileSysteming = FileSystem(),
+        commandRunner: CommandRunning = CommandRunner()
+    ) {
+        self.fileSystem = fileSystem
+        self.commandRunner = commandRunner
+    }
+
+    func codesignSignature(of xcframeworkPath: AbsolutePath) async -> String? {
```

Comment:
I'd add some tests to this controller:

- codesignSignature:
  - It returns the signature if it contains it
  - It returns `nil` if it has no signature
- codesignExtractSignature:
  - It extracts the signature if it has it
  - It errors if the XCFramework has no signature.
-